Content.Server/Attachments/AttachmentSystem.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Linq;
+using System.Reflection;
+using Content.Shared.Attachments;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Attachments;
+
+public sealed partial class AttachmentSystem : SharedAttachmentSystem
+{
+    protected override object? GetComponentFieldInfo(Type type, string field)
+    {
+        if (type.GetField(field) is { } propInfoNormal)
+            return propInfoNormal;
+        else if (type.GetField(char.ToUpper(field[0]) + field[1..]) is { } propInfoUpper) // Try harder
+            return propInfoUpper;
+        else if (type.GetFields() // Try even harder
+                     .ToList()
+                     .Find(fieldInfo => fieldInfo.HasCustomAttribute<DataFieldAttribute>()) is {} fieldInfoExisting
+                 && fieldInfoExisting.GetCustomAttribute<DataFieldAttribute>()?.Tag == field)
+        {
+
+            return fieldInfoExisting;
+        }
+        else
+        {
+            throw new ArgumentException(
+                $"Field '{field}' does not exist publicly in component type '{type}'");
+        }
+    }
+
```

Comment:
1. We have copycomp now, and component copying should only ever extend ISerializationManager.CopyTo, this is omegacursed.
2. Attachments should either use entities added to the target to store the data OR copying components and storing the original somewhere.